scripts/common.js

```diff
@@ -6,6 +8,336 @@ export let commonOptions = {
     ]
 };
 
+export function summaryFormatter(data) {
+    // removing default metrics
+    delete data.metrics['http_req_duration']
+    delete data.metrics['http_req_sending']
+    delete data.metrics['http_req_receiving']
+    delete data.metrics['http_req_blocked']
+    delete data.metrics[`http_req_duration{expected_response:true}`]
+    delete data.metrics['http_req_waiting']
+    delete data.metrics['http_reqs']
+    delete data.metrics['iteration_duration']
+    delete data.metrics['iterations']
+    delete data.metrics['vus']
+    delete data.metrics['http_req_connecting']
+    delete data.metrics['http_req_failed']
+    delete data.metrics['http_req_tls_handshaking']
+
+    data = destinations.reduce((curData, curDestinations) => {
+        return reportPerDestination(curData, curDestinations)
+    }, data)
+
+    return {
+        stdout: textSummary(data, { indent: ' ', enableColors: true }),
+    }
+}
+
+function reportPerDestination(data, destinations) {
+    data.metrics[`http_req_sending(${destinations} destinations)`] = data.metrics[`http_req_sending{scenario:sending_${destinations}_destinations}`]
+    delete data.metrics[`http_req_sending{scenario:sending_${destinations}_destinations}`]
+    data.metrics[`http_req_receiving(${destinations} destinations)`] = data.metrics[`http_req_receiving{scenario:sending_${destinations}_destinations}`]
+    delete data.metrics[`http_req_receiving{scenario:sending_${destinations}_destinations}`]
+    data.metrics[`http_req_duration(${destinations} destinations)`] = data.metrics[`http_req_duration{scenario:sending_${destinations}_destinations}`]
+    delete data.metrics[`http_req_duration{scenario:sending_${destinations}_destinations}`]
+    return data
+}
+
+export const destinations = (__ENV.DESTINATIONS || '50, 100, 150')
+    .split(',')
+    .map((curDestinations) => parseInt(curDestinations))
+
+const scenarios = destinations.reduce((accumulator, currentDestinations) => {
+    accumulator[`sending_${currentDestinations}_destinations`] = {
+        executor: 'constant-vus',
+        duration: '3m',
+        env: {SCENARIO_DESTINATIONS: currentDestinations.toString()},
+
+        vus: 1,
+        startTime: '10s',
+        gracefulStop: '10s',
+    }
+    return accumulator
+}, {})
+
+export const configs = {
+    scenarios: scenarios,
+    summaryTrendStats: ['avg', 'min', 'max', 'p(90)', 'p(95)'],
+
+    thresholds: {
+        // Intentionally empty. I'll define bogus thresholds (to generate the sub-metrics) below.
+    }
+}
+
+export function setThresholdsForScenarios(options) {
+    for (let key in options.scenarios) {
+        options.thresholds[`http_req_duration{scenario:${key}}`] = ['max>=0'];
+        options.thresholds[`http_req_receiving{scenario:${key}}`] = ['max>=0'];
+        options.thresholds[`http_req_sending{scenario:${key}}`] = ['max>=0'];
+    }
+};
+
+const countryData = `CountryName,CapitalName,CapitalLatitude,CapitalLongitude,CountryCode,ContinentName,
```

Comment:
you need to format this data. Leave only country code and coordinates for now. Rename countries to protoCountries and this one to countries.

Example:
const countries = {
    'lv': {lat: 56.945614, lng: 24.120870},
    'nl': {lat: 52.3650144, lng: 4.892851},
    'at': {lat: 48.2244617, lng: 16.326472},
    'be': {lat: 50.8610222, lng: 4.384314},
    'de': {lat: 52.5446, lng: 13.35},
    'fr': {lat: 48.8540899, lng: 2.325747},
};

Reply:
Done